### PR TITLE
Core/Quest: improve source item deletion logic for items that give quests

### PR DESCRIPTION
**Pull request by Wyrserth**
**Changes proposed:**

Items that give quests have some funky behaviors (thank you, Blizzard):

- Items that aren't Bind on Pickup are deleted when the quest is accepted if quest_template.StartItem is 0 (this already works fine before and after this PR).
- Items that start a quest whose StartItem field is the same id as said item are NOT deleted on quest accept/turn-in. This applies to [Demon Scarred Cloak](https://www.wowhead.com/item=4854) ([proof 1](https://www.wowhead.com/item=4854/demon-scarred-cloak#comments:id=234608), [proof 2](https://www.wowhead.com/quest=770/the-demon-scarred-cloak#comments:id=1056598)), [Cracked Silithid Carapace](https://www.wowhead.com/item=5877) ([proof](https://www.wowhead.com/item=5877/cracked-silithid-carapace#comments:id=883435)) and other equippable armor/weapon. This behavior leads to...
- Quests that have StartItem set to the same item id that started the quest must NOT be given to the player on quest accept, because the previous item was not deleted as per the precedent behavior. Examples: [Arena Master](https://www.wowhead.com/item=18706) ([video proof](https://youtu.be/gedm4B7Rgv8?t=230)), [Battered Hilt](https://www.wowhead.com/item=50379) ([video proof](https://youtu.be/4tEGvJvvXXk?t=4)). Notice the lack of "obtained item x" in the chat logs and how the item does not change place due to addnew+deleteold: there's no new item created, the old one is kept instead.

Long story short, this PR prevents the destruction of items that aren't supposed to be deleted ([Demon Scarred Cloak](https://www.wowhead.com/item=4854), [Pendant of Myzrael](https://www.wowhead.com/item=4614), [Noboru's Cudgel](https://www.wowhead.com/item=6196), [Cracked Silithid Carapace](https://www.wowhead.com/item=5877) and probably other useless white unique items Blizzard forgot about) and fixes a cosmetic detail on accepting two quests.

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Tests performed:** builds and works.

### CODE REVIEW

**Comment by illfated** [comment]
Looks like the **ci/circleci: nopch** issue can be solved by rebasing your PR branch on commit https://github.com/TrinityCore/TrinityCore/commit/06d8f4dc78ed731430efa5a7d4ace5571de943e0 (or current HEAD if you like).

**Comment by Wyrserth** [comment]
Rebased, thanks.

**Comment by jackpoz** [comment]
Thanks for the PR :)
